typespec-tests/tsp/server.tsp

```diff
@@ -59,8 +59,8 @@ namespace Cadl.ContosoServer {
     name: "Contoso.Sub.ContosoClient",
     service: Cadl.ContosoServer,
   })
-  @route("/contoso/{group}")
+  @route("/contoso/")
   interface ServerOp {
-    get(@path group: url): OkResponse | NoContentResponse;
+    get(@path(#{ allowReserved: true }) group: string): OkResponse | NoContentResponse;
   }
```

Comment:
0.59.0 requires either way
- `/contoso/{+group}` without `allowReserved`
- `/contoso/` with `allowReserved`. compiler or http lib would append the `group` to path

Reply:
Do we want to add another test case using URI template? (`/contoso/{+group}`)? Or cadl-ranch already has case to cover it?

Reply:
Timothee is adding a cadl-ranch test.

My local test, that `allowReserved` does not work if route written as `@route("/contoso/{+group}")`. Could be the http lib just don't want to support this (say it require we use URI template directly), or it could be a bug.

Reply:
Ok, it need to be 

```
  @route("/contoso/{+group}")
  interface ServerOp {
    get(group: string): OkResponse | NoContentResponse;
  }
```

Reply:
> Timothee is adding a cadl-ranch test.
> 
> My local test, that `allowReserved` does not work if route written as `@route("/contoso/{+group}")`. Could be the http lib just don't want to support this (say it require we use URI template directly), or it could be a bug.
> 
> Same effect on playground (Swagger having `x-ms-skip-url-encoding` or not).

This appears to be a bug in http lib.

The `getPathParamOptions` gives `undefined`. But `param.allowReserved` is `true`. Will report a bug there.

Therefore I switched to `param.allowReserved`.